core/src/test/scala/integration/kafka/api/BaseConsumerTest.scala

```diff
@@ -314,5 +315,60 @@ abstract class BaseConsumerTest extends IntegrationTestHarness with Logging {
 
     override def onComplete(offsets: util.Map[TopicPartition, OffsetAndMetadata], exception: Exception): Unit = count += 1
   }
+  protected class ConsumerAssignmentPoller(consumer: Consumer[Array[Byte], Array[Byte]]) extends ShutdownableThread("daemon-consumer-assignment", false)
+  {
+    private var partitionAssignment = scala.collection.mutable.Set[TopicPartition]()
```

Comment:
Since we always reset the assignment set as a whole, could we just tag it as volatile and remove the lock?